modules/shortcodes/instagram.php

```diff
@@ -132,6 +132,79 @@ function jetpack_instagram_embed_reversal( $content ) {
 	return $content;
 }
 
+/**
+ * List of whitelisted and sanitized parameters
+ * that can be used with the Instagram oEmbed endpoint.
+ *
+ * Those parameters can be provided via the Instagram URL, or via shortcode parameters.
+ *
+ * @see https://developers.facebook.com/docs/graph-api/reference/instagram-oembed#parameters
+ *
+ * @since 9.1.0
+ *
+ * @param string $url  URL of the content to be embedded.
+ * @param array  $atts Shortcode attributes.
+ *
+ * @return array $params Array of parameters to be used in Instagram query.
+ */
+function jetpack_instagram_get_whitelisted_parameters( $url, $atts = array() ) {
+	global $content_width;
+
+	// Any URL passed via a shortcode attribute takes precedence.
+	if ( ! empty( $atts['url'] ) ) {
+		$url = $atts['url'];
+		unset( $atts['url'] );
+	}
+
+	/*
+	 * Get URL and parameters from the URL if possible.
+	 *
+	 * We'll also clean any other query params from the URL since Facebook's new API for Instagram
+	 * embeds does not like query parameters. See p7H4VZ-2DU-p2.
+	 */
+	$parsed_url = wp_parse_url( $url );
+	if ( $parsed_url ) {
+		// Bail early if this is not an Instagram URL.
+		if ( ! preg_match( '/instagr(\.am|am\.com)/', $parsed_url['host'] ) ) {
+			return array();
+		}
+
+		$url = 'https://www.instagram.com' . $parsed_url['path'];
+
+		// If we have any parameters as part of the URL, we merge them with our attributes.
+		if ( ! empty( $parsed_url['query'] ) ) {
+			$query_args = array();
+			wp_parse_str( $parsed_url['query'], $query_args );
+
+			$atts = array_merge( $atts, $query_args );
+		}
+	}
+
+	$max_width = 698;
+	$min_width = 320;
+
+	$params = shortcode_atts(
+		array(
+			'url'         => $url,
+			'width'       => isset( $content_width ) ? $content_width : $max_width,
+			'height'      => '',
+			'hidecaption' => false,
+		),
+		$atts,
+		'instagram'
+	);
+
+	// Ensure width is within bounds.
+	$params['width'] = absint( $params['width'] );
+	if ( $params['width'] > $max_width ) {
+		$params['width'] = $max_width;
+	} elseif ( $params['width'] < $min_width ) {
+		$params['width'] = $min_width;
+	}
```

Comment:
Could alternatively do the width handling in the [`embed_defaults`](https://developer.wordpress.org/reference/hooks/embed_defaults/) filter?

Reply:
Oh, I didn't know about that filter, thank you. I think we'd want to keep this in `jetpack_instagram_get_allowed_parameters` though, to ensure we support folks using the shortcode and AMP views as well as the folks using the regular embed.